### ISSUE TITLE
Strange result for `.between` for Feb-March

### ISSUE THREAD

**Issue by forabi** [dlevs/time#33]
```ts
d.between(new Date("2024-02-27"), new Date("2024-03-02"))
{
  years: 0,
  months: 1,
  weeks: 0,
  days: -25,
  hours: 0,
  minutes: 0,
  seconds: 0,
  milliseconds: 0
}
```

**Comment by forabi** [comment]
Actually it seems to happen for any months I tried, wouldn't it make sense for the result to normalized in days?

**Comment by dlevs** [comment]
Hi!

You're right, it's a normalization thing.

This is [covered in the docs as expected behaviour](https://github.com/dlevs/duration-fns?tab=readme-ov-file#normalization). Units like "month" are ambiguous. It's impossible to know if you wanted to count it as months or days without a bit more information, so it defaults to what is safe.

To achieve what you want, you can pass the result through `.normalize()`, using the 2nd argument to specify the original reference date. Bit verbose, but that's how it works at the moment.

**Comment by ilyary** [comment]
I have a very similar issue:
```ts
const date = new Date('2025-03-20T11:42:24.649Z')

console.log(between(date, addDays(date, 29)), formatDuration(between(date, addDays(date, 29))))
// {years: 0, months: 1, weeks: 0, days: -2, hours: 0, …} '1 month -2 days'

console.log(normalize(between(date, addDays(date, 29))), formatDuration(normalize(between(date, addDays(date, 29)))))
// {years: 0, months: 1, weeks: 0, days: -2, hours: 0, …} '1 month -2 days'

console.log(
	normalize(between(date, addDays(date, 29)), date),
	formatDuration(normalize(between(date, addDays(date, 29)), date)),
	toString(normalize(between(date, addDays(date, 29)), date)),
)
// {years: 0, months: 1, weeks: 0, days: -2, hours: 0, …} '1 month -2 days' 'P1M-2D'
```

Expected behavior at least after `normalize` to have no negatives. @dlevs, `referenceDate` in `normalize` does not help

is there any way to have all not negative? I expect `normalize` does the trick, but it does not((

**Comment by dlevs** [comment]
@ilyary I agree this behaviour is odd. I've written a failing test case for it on this branch: https://github.com/dlevs/duration-fns/compare/master...issues/33-normalisation-behaviour

I won't have time to fix it myself, but I'd accept a PR.

This would be a breaking change, so it would need to bump the major version, or we could put it behind a another argument on the normalize function. I'd prefer the former, assuming that the existing behaviour isn't _ever_ useful.

We actually already have tests that assert the weird behaviour as valid:

```ts
		expect(normalize({ days: -31 }, '2020-12-01')).toEqual({
			...ZERO,
			months: -2,
			days: 30,
		});
```

I can't think of a time we'd want this behaviour, but someone needs to give it a little thought, and raise a PR.
